app/views/two_factor/tf_status.html.erb

```diff
@@ -45,4 +45,17 @@
       <% end %>
     <% end %>
   </div>
-<% end %>
\ No newline at end of file
+<% end %>
+
+<h3>Recovery</h3>
+<p>
+  Your <strong>backup code</strong> for two-factor authentication can be found below. You should save this somewhere
+  safe &mdash; if you lose access to your authenticator app, you can sign in using this code to get back into your
+  account and reconfigure 2FA.
+</p>
+
+<details>
+  <summary>Show code</summary>
+  <label for="backup-code" class="form-element">2FA backup code</label>
```

Comment:
Normally, sites give me backup codes only once (when configuring) and then don't let me see them any more. There is a potential security risk if someone can access their 2FA backup code from their account, as this allows you to effectively go from having a session cookie to full access to the account.

As a suggestion, how about adding a checkbox + button that says "I have saved this backup code." and after confirmation the backup code will no longer be displayed on this page?

Reply:
An alternative approach for consideration:

- Require a 2FA code before viewing the backup code.

This means if I leave my account signed in and unattended, no one can view my backup code unless they also have my 2FA device (such as my mobile phone).

Reply:
Done - I've required a current 2FA code before the backup code is shown.